mail-filter/sqlgrey/sqlgrey-1.8.0.ebuild

```diff
@@ -0,0 +1,252 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit systemd
+
+DESCRIPTION="A postfix policy service implementing a grey-listing policy"
+HOMEPAGE="http://sqlgrey.sourceforge.net/"
+SRC_URI="mirror://sourceforge/sqlgrey/sqlgrey-1.8%20%28stable%29/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~sparc ~x86"
+IUSE="mysql postgres +sqlite"
+REQUIRED_USE="|| ( mysql postgres sqlite )"
+
+RDEPEND="acct-user/sqlgrey
+	dev-lang/perl
+	dev-perl/DBI
+	dev-perl/Date-Calc
+	dev-perl/Net-Server
+	virtual/mailx
+	mysql? ( dev-perl/DBD-mysql )
+	postgres? ( dev-perl/DBD-Pg )
+	sqlite? ( dev-perl/DBD-SQLite )"
+DEPEND="${RDEPEND}
+	sys-apps/sed"
+
+DOCS="HOWTO FAQ README README.OPTINOUT README.PERF TODO Changelog"
+PATCHES=(
+	"${FILESDIR}/sqlgrey-1.8.0-init-openrc.patch"
+)
+
+src_install () {
+	emake gentoo-install ROOTDIR="${D}"
+	einstalldocs
+
+	systemd_dounit "${FILESDIR}/${PN}.service"
+}
+
+pkg_postinst() {
+	elog "To make use of greylisting, please update your postfix config."
+	elog
+	elog "Put something like this in ${ROOT}/etc/postfix/main.cf:"
+	elog "    smtpd_recipient_restrictions ="
+	elog "           ..."
+	elog "           check_policy_service inet:127.0.0.1:2501"
+	elog
+	elog "Remember to restart Postfix after that change. Also remember"
+	elog "to make the daemon start durig boot:"
+	elog "  rc-update add sqlgrey default"
+	elog
+	elog
+	elog "To setup SQLgrey to run out-of-the-box on your system, run:"
+	elog "emerge --config ${PN}"
+	elog
+	ewarn "Read the documentation for more info (perldoc sqlgrey) or the"
+	ewarn "included HOWTO in ${ROOT}/usr/share/doc/${PF}/"
+	ewarn
+	ewarn "If you are using MySQL >= 4.1 use \"latin1\" as charset for"
+	ewarn "the SQLgrey db"
+}
+
+pkg_config () {
+	# SQLgrey configuration file
+	local SQLgrey_CONFIG="${ROOT}/etc/sqlgrey/sqlgrey.conf"
+	local SQLgrey_DB_USER_NAME="sqlgrey"
+	local SQLgrey_DB_NAME="sqlgrey"
+
+	# Check if a password is set in sqlgrey.conf
+	local SQLgrey_CONF_PWD=""
+	if [[ -f "${SQLgrey_CONFIG}" ]]; then
+		if grep -iq "^[\t ]*db_pass[\t ]*=[\t ]*.*$" "${SQLgrey_CONFIG}"; then
+			# User already has a db_pass entry
+			SQLgrey_CONF_PWD="$(sed -n 's:^[\t ]*db_pass[\t ]*=[\t ]*\(.*\)[\t ]*:\1:gIp' "${SQLgrey_CONFIG}")"
+		else
+			SQLgrey_CONF_PWD=""
+		fi
+	else
+		ewarn "SQLgrey configuration missing. Exiting now."
+		return 0
+	fi
+
+	# Check if we need SQLgrey to configure for this system or not
+	local SQLgrey_DB_HOST="localhost"
+	local SQLgrey_KEY_INPUT="l,r,x"
+	einfo "SQLgrey database backend configuration"
+	einfo "  Please select where SQLgrey database will run:"
```

Comment:
Why not just ask for hostname, and default it to `localhost` (via `read -i ...`).